comp/process/runner/runnerimpl/runner.go

```diff
@@ -55,17 +54,10 @@ func newRunner(deps dependencies) (runner.Component, error) {
 	}
 	c.Submitter = deps.Submitter
 
-	runner := &runnerImpl{
+	return &runnerImpl{
 		checkRunner:    c,
 		providedChecks: checks,
-	}
-
-	deps.Lc.Append(fx.Hook{
```

Comment:
We prefer to use fx's lifecycles to orchestrate component startup. Why are you removing it? Is it the reason you mention elsewhere about the workloadmeta extractor taking the required port?